webapp/components/sidebar.jsx

```diff
@@ -382,11 +383,13 @@ export default class Sidebar extends React.Component {
     }
 
     sortChannelsByDisplayName(a, b) {
+        const locale = LocalizationStore.getLocale();
+
         if (a.display_name === b.display_name) {
-            return a.name.localeCompare(b.name);
+            return a.name.localeCompare(b.name, locale, {numeric: true, caseFirst: 'upper'});
```

Comment:
Sorting it by case conforms with system console team list behaviour, I discussed it with JB.